### PR TITLE
Add check op for assert valid sample in Bernoulli for contrib.distributions

### LINKED ISSUE #8583: Bernoulli's probability mass function produces output beyond its support
Bernoulli's (log) probability mass function is valid for values between 0 and 1, even though its support is only {0, 1}. This is a result of [its implementation](https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/contrib/distributions/python/ops/bernoulli.py#L150) relying on `tf.sigmoid_cross_entropy_with_logits(logits, targets)`, where `targets` does not have to be {0, 1}.

Is this intended behavior?

```python
from tensorflow.contrib import distributions as ds

x = ds.Bernoulli(p=0.5)
y = x.log_prob(0.5)

sess = tf.Session()
sess.run(y)
## -0.69314718
```

```python
print(tf.__version__)
## 1.0.1
```

### PR DESCRIPTION
**Pull request by dustinvtran**
Title. The style for asserts seemed to vary across the distribution implementations. I followed Binomial's. It implements `_maybe_assert_valid_sample` and checks via utility functions such as `embed_check_nonnegative_discrete` in `distributions_util.py`.

Fixes #8583. Feedback appreciated. After merging, I can submit a PR adding similar asserts for other discrete distributions.

### CODE REVIEW

**Comment by ebrevdo** [comment]
Can you disable the check_integer test?  I don't think it's necessary.

**Comment by dustinvtran** [comment]
Done.

**Comment by dustinvtran** [comment]
Bump. I'm happy to make changes. Afterwards I can proceed with robustifying other distributions.

**Comment by drpngx** [comment]
@ebrevdo good to go?

**Comment by ebrevdo** [comment]
LGTM

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
Mac fails with some worrisome error. Trying again.

Jenkins, test this please.

```
FAIL: //tensorflow/contrib/xla_tf_graph:xla_tf_graph_util_test (see /Volumes/Transcend/bazel_cache/13e370a18c169b19baeafefb05212b85/execroot/tensorflow-pull-requests-mac/bazel-out/local-opt/testlogs/tensorflow/contrib/xla_tf_graph/xla_tf_graph_util_test/test.log).
INFO: From Testing //tensorflow/contrib/xla_tf_graph:xla_tf_graph_util_test:
==================== Test output for //tensorflow/contrib/xla_tf_graph:xla_tf_graph_util_test:
dyld: malformed mach-o: load commands size (32744) > 32768
external/bazel_tools/tools/test/test-setup.sh: line 159: 23448 Abort trap: 6           "${TEST_PATH}" "$@"
```

**Review by drpngx — APPROVED** [review]
As ebrevdo earlier LGTM.
